src/Nethermind/Nethermind.Runner/Ethereum/EthereumRunner.cs

```diff
@@ -33,31 +33,19 @@ public async Task Start(CancellationToken cancellationToken)
 
     public async Task StopAsync()
     {
-        Stop(() => _api.SessionMonitor?.Stop(), "Stopping session monitor");
-        Stop(() => _api.SyncModeSelector?.Stop(), "Stopping session sync mode selector");
-        Task discoveryStopTask = Stop(() => _api.DiscoveryApp?.StopAsync(), "Stopping discovery app");
-        Task blockProducerTask = Stop(() => _api.BlockProducerRunner?.StopAsync(), "Stopping block producer");
-        Task peerPoolTask = Stop(() => _api.PeerPool?.StopAsync(), "Stopping peer pool");
-        Task peerManagerTask = Stop(() => _api.PeerManager?.StopAsync(), "Stopping peer manager");
-        Task blockchainProcessorTask = Stop(() => _api.MainProcessingContext?.BlockchainProcessor?.StopAsync(), "Stopping blockchain processor");
-        Task rlpxPeerTask = Stop(() => _api.RlpxPeer?.Shutdown(), "Stopping RLPx peer");
-        await Task.WhenAll(discoveryStopTask, rlpxPeerTask, peerManagerTask, peerPoolTask, blockchainProcessorTask, blockProducerTask);
+        await serviceStopper.StopAllServices();
 
-        foreach (INethermindPlugin plugin in _api.Plugins)
+        foreach (INethermindPlugin plugin in api.Plugins)
```

Comment:
Can or should we dispose plugins in parallel?

Reply:
I thought of removing this also, but that would introduce more change unnecessary to the issue. Maybe next time.